python/tank/descriptor/io_descriptor/base.py

```diff
@@ -143,7 +143,9 @@ def _validate_descriptor(cls, descriptor_dict, required, optional):
         """
         desc_keys_set = set(descriptor_dict.keys())
         required_set = set(required)
-        optional_set = set(optional)
+        # Add deny_platforms and disabled to the list of optional parameters as these are globally supported across
+        # all descriptors and are used by the environment code to check if an item is disabled.
+        optional_set = set(optional + ["deny_platforms", "disabled"])
```

Comment:
This feels like it's not encapsulated enough. Could we add properties `disabled` and one for deny_platforms to the descriptor object to maintain a clean encapsulation and then have the environment system use those? Would that be a possibility?

Reply:
Thats a good idea, adding the properties should be simple enough, but I don't think the environment code can make use of the descriptor object. I think the environment code is evaluating the env configuration before descriptors are even created.

Reply:
After discussing I will add a todo in the environment code, we are accessing a bunch of descriptor stuff, this should really be accessed via the descriptor API.

Reply:
I've added the properties and the TODO. I've added the properties but is it a good idea to do so when they are currently not in use? If this is OK I'll need to write up a test around it.